File: solutions/julia/bob/bob.jl
function bob(stimulus)

    stimulus = strip(stimulus)

    is_silence       = stimulus == ""
    is_question      = endswith(stimulus, '?') && !startswith(stimulus, '?')
    contains_letters = !isempty(filter(isletter, stimulus))
    is_yelling       = contains_letters && (uppercase(stimulus) == stimulus)

    if is_silence
        return "Fine. Be that way!"
    elseif is_question && is_yelling
        return "Calm down, I know what I'm doing!"
    elseif is_question
        return "Sure."
    elseif is_yelling
        return "Whoa, chill out!"
    else
        return "Whatever."
    end

end

File: solutions/python/bob/bob.py
def response(hey_bob):

    hey_bob = hey_bob.strip()

    is_silence       = hey_bob == ""
    is_question      = hey_bob.endswith('?') and (not hey_bob.startswith('?'))
    contains_letters = any(c.isalpha() for c in hey_bob)
    is_yelling       = contains_letters and hey_bob.upper() == hey_bob

    if is_silence:
        return "Fine. Be that way!"
    elif is_question and is_yelling:
        return "Calm down, I know what I'm doing!"
    elif is_question:
        return "Sure."
    elif is_yelling:
        return "Whoa, chill out!"
    else:
        return "Whatever."
